File: sword_to_offer/06.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

def PrintListReversingly_Iteratively(root):
    res = []
    ans = []
    if root == None:
        return []
    node = root
    while node:
        res.append(node.val)
        node = node.next
    while res:
        ans.append(res.pop())

    return ans

def PrintListReversingly_Recursively(root):
    if root != None:
        if root.next != None:
            PrintListReversingly_Recursively(root.next)
        print(root.val)

File: sword_to_offer/06.cpp
#include <iostream>
#include <stack>

using namespace std;


struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


void PrintListReversingly_Iteratively(ListNode* pHead){
    stack<ListNode*> nodes;

    ListNode* pNode = pHead;
    while(pNode != nullptr){
        nodes.push(pNode);
        pNode = pNode->next;
    }
    while(!nodes.empty()){
        pNode = nodes.top();
        cout << pNode->val << '\t';
        nodes.pop();
    }
}

void PrintListReversingly_Recursively(ListNode* pHead){
    if(pHead != nullptr){
        if(pHead->next != nullptr)
            PrintListReversingly_Recursively(pHead->next);
        cout << pHead->val << '\t';
    }
}
